pkt-meta/src/flow_table/table.rs

```diff
@@ -0,0 +1,678 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+
+use ahash::RandomState;
+use dashmap::DashMap;
+use std::borrow::Borrow;
+use std::fmt::Debug;
+use std::hash::Hash;
+use std::time::Instant;
+use tracing::{debug, error};
+
+use concurrency::sync::{Arc, RwLock, RwLockReadGuard, Weak};
+
+use crate::flow_table::thread_local_pq::{PQAction, ThreadLocalPriorityQueue};
+use crate::flow_table::{FlowInfo, FlowKey, FlowStatus};
+
+#[derive(Debug, thiserror::Error)]
+pub enum FlowTableError {
+    #[error("Invalid number of shards: {0}. Must be a power of two.")]
+    InvalidShardCount(usize),
+}
+
+type PriorityQueue = ThreadLocalPriorityQueue<FlowKey, Arc<FlowInfo>>;
+type Table = DashMap<FlowKey, Weak<FlowInfo>, RandomState>;
+
+pub struct FlowTable {
+    // TODO(mvachhar) move this to a cross beam sharded lock
+    pub(crate) table: RwLock<Table>,
+    pub(crate) priority_queue: PriorityQueue,
+}
+
+impl Default for FlowTable {
+    fn default() -> Self {
+        Self::new(1024)
+    }
+}
+
+fn hasher_state() -> &'static RandomState {
+    use std::sync::OnceLock;
+    static HASHER_STATE: OnceLock<RandomState> = OnceLock::new();
```

Comment:
I wonder if it is possible to make this `const` rather than static.  May run faster.  Not sure.